### ISSUE TITLE
Installation of CLI: adding to PATH not functional

### ISSUE THREAD

**Issue by zsiegel92** [PostgresApp/PostgresApp#390]
Hey there,

The code snippet on your installation splash page (http://postgresapp.com/documentation/cli-tools.html) didn't successfully add psql to my PATH:

`sudo mkdir -p /etc/paths.d &&
echo /Applications/Postgres.app/Contents/Versions/latest/bin | sudo tee /etc/paths.d/postgresapp`

Now, I used a more straightforward command to add psql my path temporarily:

`export PATH="/Applications/Postgres.app/Contents/Versions/latest/bin:$PATH"`

Please consider posting this on your website - if your rather complicated path-appending command fails for some users, as it did for me, it may be desirable to give a one-session fix. This is especially useful information for users setting up Heroku, who are encouraged to set up Postgres and not think about it again., and may want to be sure it's functional

**Comment by zsiegel92** [comment]
I noticed that this exact piece of code:
`PATH="/Applications/Postgres.app/Contents/Versions/latest/bin:$PATH"`


[is on the Heroku website](https://devcenter.heroku.com/articles/heroku-postgresql#local-setup)!

It seems like I'm not the only one who might think this would be useful on your site.

**Comment by jakob** [comment]
Hi!

Previously the instructions would tell users to add that line into their profile, but that seemed to be difficult, eg. some people used zsh, which has slightly different syntax, etc.

So we came up with the current method (originally suggested by #259). I thought it was simpler, since it is just a single command to execute, and it doesn't matter if you use bash or zsh. It's more complicated than I'd like, but it seems to work for most people.

I don't think that a solution that only works for the current session is a good idea. I'd assume that most people who install Postgres.app want to use it more than once.

Could it be that the issue was that you didn't reopen your terminal? The changes to paths.d only take effect when logging in.

**Comment by zsiegel92** [comment]
Hey Jakob,

Thanks so much for your response.

I agree that folks will usually want a permanent path addition. I did in
fact work this out by adding a few lines to my .profile.

I certainly did reopen my terminal, and even restarted my computer. I can't
say what the problem was with the conmand you posted.

*If other users have trouble and you want to investigate further, please
feel free to contact me at this email address.*

Thanks again!

**Comment by jakob** [comment]
Could you check the contents of `/etc/paths.d/`? What output do these two commands have?

    ls /etc/paths.d
    cat /etc/paths.d/postgresapp

Of course, if you've changed your `.profile` file it's possible that adding paths using /etc/paths just doesn't work anymore (eg. if you reset your $PATH somewhere in .profile)

**Comment by tbussmann** [comment]
the `path_helper` which is responsible for processing the content `/etc/paths.d/` is actually executed by the systemwide `/etc/profile` - maybe you want to check this as well, it might be modified. Mine is looking as following:
```
$ cat /etc/profile
# System-wide .profile for sh(1)

if [ -x /usr/libexec/path_helper ]; then
	eval `/usr/libexec/path_helper -s`
fi

if [ "${BASH-no}" != "no" ]; then
	[ -r /etc/bashrc ] && . /etc/bashrc
fi
```

**Comment by cappert** [comment]
@zsiegel92 : what shell are you using, bash, zsh?
FYI we can't see your email address to contact you further.

dupe #383

**Comment by zsiegel92** [comment]
I am using the default Mac OS X Sierra Terminal shell (bash).

I'm impressed by this community's responsiveness!

**Comment by zsiegel92** [comment]
Hi Jakob,

Here are the results of the commands you suggested:


Zachs-MacBook-Air:~ Zach$ ls /etc/paths.d

40-XQuartz TeX postgresapp

Zachs-MacBook-Air:~ Zach$ cat /etc/paths.d/postgresapp

/Applications/Postgres.app/Contents/Versions/9.5/bin


My complete .profile is copied below. The first line sets the path and all
the other commands append to it. Does that first line "reset" $PATH? Is
there something unusual about this?



export
PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/smlnj-110.75/bin


##

# Your previous /Users/Zach/.profile file was backed up as
/Users/Zach/.profile.macports-saved_2012-11-15_at_14:18:40

##


# MacPorts Installer addition on 2012-11-15_at_14:18:40: adding an
appropriate PATH variable for use with MacPorts.

export PATH=/opt/local/bin:/opt/local/sbin:$PATH

# Finished adapting your PATH environment variable for use with MacPorts.



##

# Your previous /Users/Zach/.profile file was backed up as
/Users/Zach/.profile.macports-saved_2013-09-22_at_04:38:47

##


# MacPorts Installer addition on 2013-09-22_at_04:38:47: adding an
appropriate PATH variable for use with MacPorts.

export PATH=/opt/local/bin:/opt/local/sbin:$PATH

# Finished adapting your PATH environment variable for use with MacPorts.



# MacPorts Installer addition on 2016-12-12_at_08:46:30: adding an
appropriate PATH variable for use with MacPorts.

export PATH="/opt/local/bin:/opt/local/sbin:$PATH"

# Finished adapting your PATH environment variable for use with MacPorts.



# added by Anaconda3 4.3.0 installer

export PATH="/Users/Zach/anaconda/bin:$PATH"


# Setting PATH for Python 3.6

# The original version is saved in .profile.pysave

PATH="/Library/Frameworks/Python.framework/Versions/3.6/bin:${PATH}"

export PATH


# Setting PATH for Postgres

export PATH=/Applications/Postgres.app/Contents/Versions/latest/bin:$PATH

**Comment by zsiegel92** [comment]
I have the same output:

cat /etc/profile

# System-wide .profile for sh(1)

if [ -x /usr/libexec/path_helper ]; then

eval `/usr/libexec/path_helper -s`

fi

if [ "${BASH-no}" != "no" ]; then

[ -r /etc/bashrc ] && . /etc/bashrc

fi

**Comment by jakob** [comment]
@zsiegel92 Yes, the first line in your `.profile` resets your `$PATH`. Usually people just add to the path by appending stuff in front or in the back like this:
```bash
export PATH=/high/priority/path:$PATH
export PATH=$PATH:/high/priority/path
```

If you don't include the `$PATH` variable, you override the previous value.

So when `bash` starts, it does the following:

1) Execute `/etc/profile` -> uses path_helper to set the $PATH according to `/etc/paths` and `/etc/paths.d`
2) Execute you `~/.profile` which overwrites $PATH

That's why the path in `/etc/paths.d/postgresapp` was ignored.

**Comment by tbussmann** [comment]
Thank you for providing these details. Here we can find the root of the issue. In your .profile the PATH is overwritten as you suspected. In `export PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/smlnj-110.75/bin` the previously set path is overwritten. You should change this in something like `export PATH=$PATH:/usr/local/smlnj-110.75/bin` - how did you install smlnj? I had a quick look into the latrest .pkg from http://www.smlnj.org and could not find any profile modification in the postinstall script.

Further: 
```
$ cat /etc/paths.d/postgresapp
/Applications/Postgres.app/Contents/Versions/9.5/bin
```
this seems a bit strange. Here version `9.5` is specified explicitly whereas the command from the documentation would add `latest` which is a symlink to the active version.

**Comment by zsiegel92** [comment]
Thank you so much, Jakob. That helps me a lot with understanding my
computer.

I suppose it may not be worth changing the instructions on your page to
accommodate people with this situation, unless you get many more emails
from other users with this same problem.

Best wishes,
Zach
